app/src/test/java/fi/bitrite/android/ws/util/ToolsTest.java

```diff
@@ -0,0 +1,130 @@
+package fi.bitrite.android.ws.util;
+
+import android.app.Application;
+import android.os.Build;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.android.controller.ActivityController;
+import org.robolectric.annotation.Config;
+
+import java.util.Locale;
+
+import fi.bitrite.android.ws.ui.MainActivity;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(qualifiers = "fr-rFR")
+public class ToolsTest {
+
+    private static Application application;
+    @Before
+    public void setUp() {
+        application = RuntimeEnvironment.application;
+    }
+
+    @Test
+    @Config(sdk = Build.VERSION_CODES.N)
+    public void getLocale_noContext() {
+        final ActivityController<MainActivity> controller =
+                Robolectric.buildActivity(MainActivity.class);
+        controller.setup();
+
+        // Get Locale for current setup. Should be fr-FR as specified in Test's config
+        final Locale locale = Tools.getLocale(null);
+
+        Assert.assertThat("If no context supplied, should always be the same", locale,
+                is(equalTo(Locale.getDefault())));
+
+        // Resets the qualifier to set the Locale to German
+        RuntimeEnvironment.setQualifiers("de-rDE");
+        controller.configurationChange();
+
+        // Get Locale for changed setup. Should be de-DE now
+        final Locale localeAfterChange = Tools.getLocale(null);
+
+        Assert.assertThat("If no context supplied, should always be the same", locale,
```

Comment:
Did you intend to not check whether they are different (as in the other test cases)?

Reply:
Yes, that was intentional, which is why the "reason" string at the beginning of the assertThat is the same.

When no Context is supplied, Locale.getDefault is used which stays the same during the whole app runtime; that’s what the test is for: Showing that this is what happens.

Reply:
:+1: